api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -159,6 +167,50 @@ private void validateStopDatetime(Visit visit, Visit otherVisit, Errors errors)
 			errors.rejectValue("stopDatetime", "Visit.visitCannotContainAnotherVisitOfTheSamePatient", messageBuilder
 			        .toString());
 		}
-		
+
+	}
+
+	private void validateStartDateTimeToBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
+			if (visit.getStartDatetime() != null && patient.getBirthdate() != null && !patient.getBirthdateEstimated()
+					&& visit.getStartDatetime().before(patient.getBirthdate())) {
+				errors.rejectValue("startDatetime", "Visit.startDateCannotBeBeforeBirthdate",
+						"This visit has a start date before birth date of the patient.");
+			}
+		}
+	}
+
+	private void validateStartDateTimeToEstimatedBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
+			if (visit.getStartDatetime() != null && patient.getBirthdate() != null && patient.getBirthdateEstimated()) {
+
+				Calendar calendar = Calendar.getInstance();
+				calendar.setTime(patient.getBirthdate());
+
+				long ageInMillis = Calendar.getInstance().getTimeInMillis() - calendar.getTimeInMillis();
+				long ageInYears = (ageInMillis / 86400000) / 365;
+				int ageInYearsInt = (int) ageInYears;
```

Comment:
@wluyima Will it be precise enough if I do something like this:
calendar.add(Calendar.DAY_OF_YEAR, -ageDaysInt);
instead of substracting years? ageDayisInt is converted ageDays which is ageInMillis / 86400000